Source/Predication/SmartBlockPredicator.cs

```diff
@@ -328,7 +328,7 @@ Variable FreshPredicate(ref int predCount)
               var condPred = FreshPredicate(ref predCount);
               predMap[block.Item1] = condPred;
               defMap[block.Item1] = condPred;
-              var headerIterator = headerDominance.GetEnumerator();
+              using var headerIterator = headerDominance.GetEnumerator();
```

Comment:
What is the difference in behavior as a result of this change?

Reply:
I'm guessing no practical difference, but now the method satisfies the contract of `GetEnumerator`, which is that it should be disposed after it's used. This isn't required when using `foreach` to enumerate, and I'm not sure why foreach is not used here.